pyrealm/demography/t_model_functions.py

```diff
@@ -145,6 +145,260 @@ def calculate_sapwood_masses(
+    costs for foliage  (:math:`R_f`), fine roots  (:math:`R_r`) and sapwood
+    (:math:`R_s`).
+
+    .. math::
+        P_{net} = y (P - R_m) = y (P - W_{\cdot s} r_s - \zeta \sigma W_f r_r - P r_f)
+
+    Note that this differs from Equation 13 of :cite:t:`Li:2014bc`, which removes foliar
+    respiration from potential GPP. This approach is equivalent but allows the foliar
+    respiration to vary between plant functional types.
+
+    Args:
+        yld: The yield proportion.
+        whole_crown_gpp: The total GPP for the crown.
+        foliar_respiration: The total foliar respiration.
+        fine_root_respiration: The total fine root respiration
+        sapwood_respiration: The total sapwood respiration.
+    """
+
+    return yld * (
+        whole_crown_gpp
+        - foliar_respiration
+        - fine_root_respiration
+        - sapwood_respiration
+    )
+
+
+def calculate_foliage_and_fine_root_turnover(
+    sla: Series,
+    zeta: Series,
+    tau_f: Series,
+    tau_r: Series,
+    foliage_mass: Series,
+) -> Series:
+    r"""Calculate turnover costs.
+
+    This function calculates the costs associated with the turnover of fine roots and
+    foliage. This is calculated from the total foliage mass of individuals
+    (:math:`W_f`), along with the specific leaf area (:math:`\sigma`) and fine root mass
+    to foliar area ratio (:math:`\zeta`) and the turnover times of foliage
+    (:math:`\tau_f`) and fine roots (:math:`\tau_r`) of the plant functional type
+    :cite:p:`{see Equation 15, }Li:2014bc`.
+
+    .. math::
+
+        T = W_f \left( \frac{1}{\tau_f} + \frac{\sigma \zeta}{\tau_f} \right)
+
+    Args:
+        sla: The specific leaf area
+        zeta: The ratio of fine root mass to foliage area.
+        tau_f: The turnover time of foliage
+        tau_r: The turnover time of fine roots
+        foliage_mass: The foliage mass
+    """
+
+    return foliage_mass * ((1 / tau_f) + (sla * zeta / tau_r))
+
+
+def calculate_growth_increments(
+    rho_s: Series,
+    a_hd: Series,
+    h_max: Series,
+    lai: Series,
+    ca_ratio: Series,
+    sla: Series,
+    zeta: Series,
+    npp: Series,
+    turnover: Series,
+    dbh: Series,
+    height: Series,
+) -> tuple[Series, Series, Series]:
+    r"""Calculate growth increments.
+
+    Given an estimate of net primary productivity (:math:`P_{net}`), less associated  
+    turnover costs (:math:`T`), the remaining productivity can be allocated to growth
+    and hence estimate resulting increments in:
+    
+    * the stem diameter (:math:`\Delta D`),
+    * the stem mass (:math:`\Delta W_s`), and 
+    * the foliar mass (:math:`\Delta W_f`). 
+        
+        
+    The stem diameter increment can be calculated using the available productivity for
+    growth and the rates of change in stem (:math:`\textrm{d}W_s / \textrm{d}t`) and
+    foliar masses (:math:`\textrm{d}W_f / \textrm{d}t`): 
+
+    .. math::
+
```

Comment:
Is there a citation available for this approach?

Reply:
It's Lie et al. again